spark/src/main/java/org/apache/iceberg/spark/data/SparkOrcWriter.java

```diff
@@ -261,9 +268,14 @@ public void addValue(int rowId, int column, SpecializedGetters data,
         output.isNull[rowId] = true;
       } else {
         output.isNull[rowId] = false;
-        ((DecimalColumnVector) output).vector[rowId].set(
-            HiveDecimal.create(data.getDecimal(column, precision, scale)
-                .toJavaBigDecimal()));
+
+        Decimal decimal = data.getDecimal(column, precision, scale);
+        Preconditions.checkArgument(scale == decimal.scale(),
+            "Cannot write value as decimal(%s,%s), wrong scale: %s", precision, scale, decimal);
+        Preconditions.checkArgument(decimal.precision() <= precision && precision <= 38,
+            "Cannot write value as decimal(%s,%s), invalid precision: %s", precision, scale, decimal);
```

Comment:
This check seems redundant to me. If we are already passing our expected precision and scale to `data.getDecimal()`, wont the scale and precision of the returned decimal always match?